### PR TITLE
Add format_as_dict, remove asdict method

### PR DESCRIPTION
**Pull request by lopuhin**
Apart from consistency with other formatters, another benefit is that it's possible to use format_as_dict on parts of explanation too, which can be useful when sending data to other clients.

### CODE REVIEW

**Comment by kmike** [comment]
Could you please add it to generated docs (https://github.com/TeamHG-Memex/eli5/blob/master/docs/source/autodocs/formatters.rst)?

**Comment by lopuhin** [comment]
> Could you please add it to generated docs

Done (messed up formatting in the first commit). I put it last cause it's probably not the first thing you want from formatters, or it's better to keep the lexicographic order?

**Comment by kmike** [comment]
>  I put it last cause it's probably not the first thing you want from formatters, or it's better to keep the lexicographic order?

Makes sense, I think that's perfect.
